### PR TITLE
random homework

### PR DESCRIPTION
**Pull request by NMandel527**
Screenshot 2024-03-13 001503

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -9,9 +10,16 @@ repositories {
     mavenCentral()
 }
 
+
+checkstyle {
+    configFile = '.guthub/workflows/checkstyles.xml' as File
```
Comment (andrewoid) [inline build.gradle]: ".github"

File: src/main/java/mandel/fallingsand/Main.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+public class Main {
+    public static void main(String[] args) {
+        Sand sand = new Sand(50, 10);
+        sand.randomSand(50);
+        Scanner input = new Scanner(System.in);
+        try {
+            sand = new Sand(Integer.parseInt(args[0]), Integer.parseInt(args[1]));
+            sand.randomSand(Integer.parseInt(args[2]));
+        } catch (Exception ex) {
+            System.out.println(ex.getMessage());
```
Comment (andrewoid) [inline src/main/java/mandel/fallingsand/Main.java]: e.printStackTrace()

File: src/main/java/mandel/fallingsand/Main.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.Scanner;
+
+public class Main {
+    public static void main(String[] args) {
+        Sand sand = new Sand(50, 10);
+        sand.randomSand(50);
+        Scanner input = new Scanner(System.in);
+        try {
+            sand = new Sand(Integer.parseInt(args[0]), Integer.parseInt(args[1]));
+            sand.randomSand(Integer.parseInt(args[2]));
```
Comment (andrewoid) [inline src/main/java/mandel/fallingsand/Main.java]: You do this on line 8-9.
Why are you doing it again?

File: src/main/java/mandel/fallingsand/Sand.java [hunk 4]
```diff
@@ -25,37 +36,57 @@ public String toString()
...
+                    }
                 }
             }
         }
     }
-}
+
+    public void randomSand(int n) {
+        int x;
+        int y;
```
Comment (andrewoid) [inline src/main/java/mandel/fallingsand/Sand.java]: You can move this into the loop

File: src/main/java/mandel/fallingsand/Main.java (hunk 3 above)
Comment (NMandel527) [reply src/main/java/mandel/fallingsand/Main.java]: I thought we were supposed to use command line, but it wouldn't work if I just did it inside the try. Should I get rid of the whole try-catch and leave lines 8-9?

Comment (andrewoid) [reply src/main/java/mandel/fallingsand/Main.java]: Yes.

Comment (NMandel527) [reply src/main/java/mandel/fallingsand/Main.java]: I deleted the extra code, but now my build is failing again

File: build.gradle (hunk 1 above)
Comment (andrewoid) [reply build.gradle]: "checkstyle.xml"

Comment (NMandel527) [reply build.gradle]: Great it worked, thank you so much

**Comment by andrewoid** [comment]
post a screenshot

File: src/main/java/mandel/fallingsand/Main.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+public class Main {
+    public static void main(String[] args) {
+        Sand sand = new Sand(50, 10);
+        Scanner input = new Scanner(System.in);
+        sand.randomSand(50);
+        while (true) {
+            System.out.println("Enter");
+            String enter = input.nextLine();
+            if (enter.isEmpty()) {
```
Comment (andrewoid) [inline src/main/java/mandel/fallingsand/Main.java]: Can you change it so that it goes until all the sand settles?

Comment (NMandel527) [reply src/main/java/mandel/fallingsand/Main.java]: It should settle all the way down each time the user presses enter? So before they press enter again, all the sand from the previous enter is already settled? Or should it just settle all the way after they finish pressing enter?

Comment (andrewoid) [reply src/main/java/mandel/fallingsand/Main.java]: settle all the way down, don't even bother having them press enter.

Comment (NMandel527) [reply src/main/java/mandel/fallingsand/Main.java]: Oh ok will try that

Comment (NMandel527) [reply src/main/java/mandel/fallingsand/Main.java]: Hi Professor Schwimmer,
I committed my most updated code, but it's not showing up on this page. It's in the repository and when I try to commit again it says there's nothing new to add, but it's not showing up here and it's not going through checkstyles. Are you able to see the code or is there something else I should do to try to re-push it?
Thank you

Comment (andrewoid) [reply src/main/java/mandel/fallingsand/Main.java]: You committed and then pushed?

Comment (NMandel527) [reply src/main/java/mandel/fallingsand/Main.java]: Yes

File: src/main/java/mandel/fallingsand/Main.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
+package mandel.fallingsand;
+
+
+public class Main {
+    public static void main(String[] args) {
+        Sand sand = new Sand(50, 10);
+        sand.randomSand(50);
+        boolean keepFalling = true;
+        while (keepFalling) {
```
Comment (andrewoid) [inline src/main/java/mandel/fallingsand/Main.java:9]: You know you need to keepFalling if the Strings from toString() before and after the fall are equal.

Comment (NMandel527) [reply src/main/java/mandel/fallingsand/Main.java:9]: Thank you, just changed it. Did the way that I did it not work properly, or is it just that this way is better and shorter?

Comment (andrewoid) [reply src/main/java/mandel/fallingsand/Main.java:9]: Better and shorter

Comment (NMandel527) [reply src/main/java/mandel/fallingsand/Main.java:9]: Got it, thank you

Comment (andrewoid) [reply src/main/java/mandel/fallingsand/Main.java:9]: Try to minimize the amount of times you call toString

File: src/main/java/mandel/fallingsand/Sand.java [hunk 7]
```diff
@@ -25,37 +36,56 @@ public String toString()
...
             }
         }
     }
-}
+
+    public void randomSand(int n) {
+        for (int i = 0; i < n; i++) {
+            int length = field.length - 1;
+            int y = random.nextInt(length);
+            int x = random.nextInt(field[length].length - 1);
```
Comment (andrewoid) [inline src/main/java/mandel/fallingsand/Sand.java]: -1 means that you won't get the last row/column

File: src/main/java/mandel/fallingsand/Sand.java [hunk 8]
```diff
@@ -25,37 +36,56 @@ public String toString()
...
+                    }
                 }
             }
         }
     }
-}
+
+    public void randomSand(int n) {
+        for (int i = 0; i < n; i++) {
+            int length = field.length - 1;
```
Comment (andrewoid) [inline src/main/java/mandel/fallingsand/Sand.java]: You forgot this one.

File: src/main/java/mandel/fallingsand/Sand.java [hunk 9]
```diff
@@ -25,37 +36,56 @@ public String toString()
...
             }
         }
     }
-}
+
+    public void randomSand(int n) {
+        for (int i = 0; i < n; i++) {
+            int length = field.length;
+            int y = random.nextInt(length);
+            int x = random.nextInt(field[length - 1].length);
```
Comment (andrewoid) [inline src/main/java/mandel/fallingsand/Sand.java]: `field[y].length`

**Review by andrewoid — APPROVED** [review]

**Pull request merged**